app/core/src/main/java/com/fsck/k9/controller/push/AlarmPermissionManager.kt

```diff
@@ -0,0 +1,29 @@
+package com.fsck.k9.controller.push
+
+import android.content.Context
+import android.os.Build
+import com.fsck.k9.helper.AlarmManagerCompat
+
+/**
+ * Checks whether the app can schedule exact alarms.
+ */
+internal interface AlarmPermissionManager {
+    fun canScheduleExactAlarms(): Boolean
```

Comment:
This is exposing logic to the call site without instructions how to deal with the result. I would suggest to rename this, for example to `hasAlarmPermission`. In case it doesn't have that permission it needs to ask for it by registering a listener.

Reply:
Or emphasize that this is about `Exact` alarms specifically by addthing this to the name.

Reply:
> This is exposing logic to the call site without instructions how to deal with the result. 

Could this be solved with a kdoc comment?

> I would suggest to rename this, for example to `hasAlarmPermission`. In case it doesn't have that permission it needs to ask for it by registering a listener.

I had this as `hasAlarmPermission()` initially. However, since older Android versions don't know about the permission, it wouldn't be entirely correct to return `true` in that case. We could invert the logic and call it `isAlarmPermissionMissing()`.

> Or emphasize that this is about Exact alarms specifically by addthing this to the name.

Do you mean add it to the interface name, i.e. `ExactAlarmPermissionManager`? Does that add any value or just a word? There's no other relevant alarm permission.

Reply:
I think `kdoc` would be sufficient.